### ISSUE TITLE
java.util.ConcurrentModificationException: java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by ngoni** [transistorsoft/background-geolocation-lt#78]
## Your Environment
* Plugin version: 1.0.4
* Platform: Android
* OS version: Android 9
* Device manufacturer / model: W30
* Plugin config

```
final TSConfig config = TSConfig.getInstance(context);

  // Configure Foreground Service notification
  final TSConfig.Notification notification = buildNotification();

  // build custom fields for the API
  JSONObject params = buildHttpParams();

  // Configure the SDK:
  config.updateWithBuilder()
          .setDebug(false)
          .setLogLevel(3)
          .setLocationUpdateInterval(5000L)
          .setFastestLocationUpdateInterval(5000L)
          .setForegroundService(true)
          .setStopOnTerminate(false)
          .setStartOnBoot(true)
          .setEnableHeadless(false)
          .setNotification(notification)
          .setAutoSync(true)
          .setAutoSyncThreshold(0)
          .setBatchSync(true)
          .setMaxBatchSize(100)
          .setParams(params)
          .commit();
```

## Context

The application crashed in production, I don't have data on how to replicate the crash.

## Debug logs

```
android.util.ArrayMap.put ArrayMap.java:523
android.os.BaseBundle.putBoolean BaseBundle.java:570
com.transistorsoft.locationmanager.location.TSLocationManager.buildTSLocation TSLocationManager.java:27
com.transistorsoft.locationmanager.location.TSLocationManager.onSingleLocationResult TSLocationManager.java:353
com.transistorsoft.locationmanager.service.LocationRequestService.a LocationRequestService.java:352
com.transistorsoft.locationmanager.service.LocationRequestService.a LocationRequestService.java
com.transistorsoft.locationmanager.service.LocationRequestService$a.run LocationRequestService.java:18
java.util.concurrent.ThreadPoolExecutor.runWorker ThreadPoolExecutor.java:1167
java.util.concurrent.ThreadPoolExecutor$Worker.run ThreadPoolExecutor.java:641
java.lang.Thread.run Thread.java:764
```

**Comment by christocracy** [comment]
Might be fixed in latest `1.3.0`

**Comment by ngoni** [comment]
@christocracy  i'm experiencing a similar crash with `1.3.0`

```
android.util.ArrayMap.put ArrayMap.java:523
android.os.BaseBundle.putString BaseBundle.java:629
com.transistorsoft.locationmanager.location.TSLocationManager.onSingleLocationResult TSLocationManager.java:227
com.transistorsoft.locationmanager.service.LocationRequestService.a LocationRequestService.java:353
com.transistorsoft.locationmanager.service.LocationRequestService.a LocationRequestService.java
com.transistorsoft.locationmanager.service.LocationRequestService$a.run LocationRequestService.java:18
java.util.concurrent.ThreadPoolExecutor.runWorker ThreadPoolExecutor.java:1162
java.util.concurrent.ThreadPoolExecutor$Worker.run ThreadPoolExecutor.java:636
java.lang.Thread.run Thread.java:764
```

**Comment by christocracy** [comment]
Show me this line from the `adb logcat` with your app booting after being terminated:
```
09:58:12.158 26942 26963 I TSLocationManager: [c.t.l.adapter.TSConfig print]
06-09 09:58:12.158 26942 26963 I TSLocationManager: ╔═════════════════════════════════════════════
06-09 09:58:12.158 26942 26963 I TSLocationManager: ║ TSLocationManager version: 3.1.6 (362)
06-09 09:58:12.158 26942 26963 I TSLocationManager: ╠═════════════════════════════════════════════
06-09 09:58:12.158 26942 26963 I TSLocationManager: ╟─ Google Pixel 3a @ 10 (react-native)
```

**Comment by ngoni** [comment]
```
2020-06-10 17:55:23.809  I/TSLocationManager: [c.t.l.adapter.TSConfig print] 
    ╔═════════════════════════════════════════════
    ║ TSLocationManager version: 3.1.2 (358)
    ╠═════════════════════════════════════════════
    ╟─ samsung SM-N960F @ 10 (native)
    {
```

**Comment by ngoni** [comment]
@christocracy is this resolved in `1.4.0`?

**Comment by christocracy** [comment]
I don't know.  I've never been able to reproduce it.
